src/Chapter4.hs

```diff
@@ -628,8 +664,9 @@ Can you implement a monad version of AND, polymorphic over any monad?
 
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
+
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM xm ym = (ym >>= (\y -> pure (&& y))) <*> xm
```

Comment:
The ordinary `&&` is lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: 


With `Applicative` you are not able to implement a lazier version of `andM`.